pkg/controller/agent/fleet.go

```diff
@@ -265,6 +285,30 @@ func maybeReconcileFleetEnrollment(params Params, result *reconciler.Results) En
 	return token
 }
 
+func getKibanaBasePath(ctx context.Context, client k8s.Client, kibanaNSN types.NamespacedName) (string, error) {
+	var kb v1.Kibana
+	err := client.Get(ctx, kibanaNSN, &kb)
+	if err != nil {
+		return "", err
+	}
+
+	if kb.Spec.Config == nil {
+		return "", nil
+	}
+
+	kbucfgConfig, err := ucfg.NewFrom(kb.Spec.Config.Data)
```

Comment:
I think that one downside of that approach is that we are reading the _expected_ config, not necessarily the one currently applied in the Kibana config Secret.

Reply:
That is a great point ! We probably should look at the secret which would be the current applied config. I can make that change

Reply:
@barkbay but isn't the source of truth the kibana crd? @kvalliyurnatt and I were discussing and I'm not seeing much benefit of reading the secret vs the crd. Don't we treat the crd as the source of truth in nearly all of the controller, as we're talking eventual consistency here, right? Are there other places in the controller where we treat the secret as what should be applied across the stack? I'm trying to ensure we are consistent in how we handle this type of thing. Is there a situation you had in mind where one is better than the other?

Reply:
> @barkbay but isn't the source of truth the kibana crd

I consider the custom resources as the _desired_ state. The _current_ state can be different. It can be a temporary difference, the controller is making some progress and is implementing the desired state, or the controller can be "stuck" (bug in the controller, issue reaching the k8s api, problem in the Kibana resource itself..).

> [...] I'm not seeing much benefit of reading the secret

If the Kibana controller can't reconcile the configuration we may use an incorrect view of the configuration, and then we may also fail to reconcile the Agent resource. As mention before it can be a temporary failure (Agent controller is faster than Kibana controller), or it can a permanent one (bug in the Kibana controller, error in the Kibana resource...). In all the cases it can be confusing to see Agent controller errors while the root cause could be the Kibana controller, this is the kind of situation I wanted to avoid by challenging the proposed implementation.

Reply:
I am not strongly leaning towards one implementation over the other, I have made the changes to read form secret, but @naemono  let me know if you feel strongly  to go with reading from the CR

I think this point below makes me lean more towards reading from the secret.
>  In all the cases it can be confusing to see Agent controller errors while the root cause could be the Kibana controller, this is the kind of situation I wanted to avoid by challenging the proposed implementation.